### PR TITLE
Added GH Actions summary

### LINKED ISSUE #158: Add a GH Actions summary
### Is your enhancement related to a problem? Please describe.

Update the plugin to use the GH Actions Report summary used in https://github.com/10up/cypress-wp-utils/pull/63. It's a very nice presentation of the test results.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by faisal-alvi**
### Description of the Change
This PR utilizes [the new Markdown support](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/) of GH Actions summary (suggested by @jeffpaul) to include the test result in the job summary: https://github.com/10up/cypress-wp-utils/actions/runs/2319016114

To generate the markdown report, this PR uses `mochawesome` reporter. The reporter generates JSON and HTML reports. The report is pretty and solid:

image

Closes #158

### How to test the Change

Check the GitHub Action [result](https://github.com/10up/simple-local-avatars/actions/runs/3638853404), it should display the Cypress test result summary.

### Credits

Props @faisal-alvi @jeffpaul 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -31,3 +31,21 @@ jobs:
       run: npm run env:start
     - name: Test
       run: npm run cypress:run
+    - name: Update summary
```
Comment (iamdharmesh) [inline .github/workflows/cypress.yml:34]:
```suggestion
    - name: Update summary
      if: ${{ always() }}
```

**Review by iamdharmesh — CHANGES_REQUESTED** [review]
I think we have to add `if: ${{ always() }}` to the **Update summary** step otherwise it will only run when the E2E test succeeds and will be skipped for test failures.

**Review by iamdharmesh — APPROVED** [review]
Thanks a lot for adding this @faisal-alvi. LGTM 🚀

**Comment by jeffpaul** [comment]
@10up/open-source-practice I'd love to see this similarly replicated across our other projects as well as a note on how to do so for future projects added to the Open Source Best Practices... please and thanks!

**Pull request merged**